### PR TITLE
upgrade to quarkus 3 and jakarta

### PR DESCRIPTION
**Pull request by snasib**

### CODE REVIEW

**Comment by ranjanashish** [comment]
@snasib Please update the PR with the latest Quarkus 3 release (3.0.0.Alpha4 as of this writing) and also resolve conflicts.

File: pom.xml [hunk 1]
```diff
@@ -22,8 +22,8 @@
         <maven.compiler.target>11</maven.compiler.target>
         <project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
         <project.reporting.outputEncoding>UTF-8</project.reporting.outputEncoding>
-        <quarkus.version>2.14.3.Final</quarkus.version>
-        <org.jooq.version>3.16.11</org.jooq.version>
+        <quarkus.version>3.0.0.Alpha2</quarkus.version>
+        <org.jooq.version>3.17.6</org.jooq.version>
```
Comment (ranjanashish) [inline pom.xml:26]: IMO we should take JOOQ version upgrade in another PR.

P.S. JOOQ 3.17 requires Java 17 as minimum Java version.

**Review by ranjanashish — CHANGES_REQUESTED** [review]

File: pom.xml (hunk 1 above)
Comment (snasib) [reply pom.xml]: Yes, you are right [verions](https://www.jooq.org/download/versions#oss). `3.16.11` it is then

**Review by ranjanashish — APPROVED** [review]

**Comment by ranjanashish** [comment]
@snasib I have approved the PR. I will merge it right after making a patch release on main.

**Comment by apatrida** [comment]
@ranjanashish And that merge?

**Comment by ranjanashish** [comment]
@apatrida @snasib Apologies for the delay on this. Give me 2-3 more days. I will merge this asap.

**Comment by sandrasap** [comment]
@ranjanashish Are there any updates on the merge?

**Comment by snasib** [comment]
It has been 3 months @ranjanashish. We seriously need to move on with this. What is stopping?

**Comment by ranjanashish** [comment]
We had an issue with the release workflow in CI. It was fixed last week but I did not get time to merge this PR after that fix. Apologies for the delay. I have merged the PR now.

**Comment by ranjanashish** [comment]
New alpha release for Quarkus 3 - https://github.com/quarkiverse/quarkus-jooq/releases/tag/2.0.0.Alpha1

**Pull request merged**